src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+/*
+ * Youtube video plugin
+ */
+
+require("../../../html5-common/js/utils/InitModules/InitOO.js");
+require("../../../html5-common/js/utils/InitModules/InitOOUnderscore.js");
+require("../../../html5-common/js/utils/InitModules/InitOOHazmat.js");
+require("../../../html5-common/js/utils/constants.js");
+(function(_, $) {
+ 
+  var pluginName = "ooyalaYoutubeVideoTech";
+  var player;
+  var youtubePlayer;
+  var youtubeVideoContainer;
+  var element;
+  var youtubeID;
+  var playerNotready = true;
+  var playerReady = false;
+  var bitrateFlag = true;
+  var javascriptCommandQueue = [];
+  var qualities = [];
+
+  /**
+   * @class OoyalaYoutubeVideoFactory
+   * @classdesc Factory for creating video player objects for youtube videos.
+   * @property {string} name The name of the plugin
+   * @property {object} encodings An array of supported encoding types (ex. OO.VIDEO.ENCODING.MP4)
+   * @property {object} features An array of supported features (ex. OO.VIDEO.FEATURE.BITRATE_CONTROL)
```

Comment:
We use [] appended to the type to denote arrays in our jsdocs. See:

http://usejsdoc.org/tags-type.html

For example:
* @returns {string[]} A list of all ad managers that have been unregistered.